addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -303,6 +303,10 @@ export class HtmlField extends Component {
             this.currentEditingValue = value;
             await this.props.update(value);
         }
+        const sel = document.getSelection();
+        if (!(this._isDirty() && sel.isCollapsed)) {
```

Comment:
This is the wrong fix IMO. `updateValue` is called only by `commitChanges`, on the condition that `this._isDirty() || urgent`. This means `!this._isDirty()` will only be the case when doing an urgent save. Why do we need to remove the selection when urgent-saving? The bug is supposed to occur when discarding.

This indicates that there's another issue at play, where we end up calling `updateValue` with a non-dirty document when discarding. That doesn't seem normal to me and should really be what this addresses.